### PR TITLE
Add recipe for labmaze

### PR DESCRIPTION
**Pull request by traversaro**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by traversaro** [comment]
Windows is failing with:
~~~
Using pip 22.3.1 from %PREFIX%\lib\site-packages\pip (python 3.9)
Non-user install because user site-packages disabled
Ignoring indexes: https://pypi.org/simple
Created temporary directory: C:\Users\VssAdministrator\AppData\Local\Temp\pip-build-tracker-zi5yv9ck
  File "C:\bld\labmaze_1670963390004\_h_env\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 231, in _prepare
    dist = self._prepare_distribution()
  File "C:\bld\labmaze_1670963390004\_h_env\lib\site-packages\pip\_internal\resolution\resolvelib\candidates.py", line 308, in _prepare_distribution
    return preparer.prepare_linked_requirement(self._ireq, parallel_builds=True)
  File "C:\bld\labmaze_1670963390004\_h_env\lib\site-packages\pip\_internal\operations\prepare.py", line 491, in prepare_linked_requirement
    return self._prepare_linked_requirement(req, parallel_builds)
  File "C:\bld\labmaze_1670963390004\_h_env\lib\site-packages\pip\_internal\operations\prepare.py", line 577, in _prepare_linked_requirement
    dist = _get_prepared_distribution(
  File "C:\bld\labmaze_1670963390004\_h_env\lib\site-packages\pip\_internal\operations\prepare.py", line 69, in _get_prepared_distribution
    abstract_dist.prepare_distribution_metadata(
  File "C:\bld\labmaze_1670963390004\_h_env\lib\contextlib.py", line 126, in __exit__
    next(self.gen)
  File "C:\bld\labmaze_1670963390004\_h_env\lib\site-packages\pip\_internal\operations\build\build_tracker.py", line 124, in track
    self.remove(req)
  File "C:\bld\labmaze_1670963390004\_h_env\lib\site-packages\pip\_internal\operations\build\build_tracker.py", line 109, in remove
    os.unlink(self._entry_path(req.link))
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\VssAdministrator\\AppData\\Local\\Temp\\pip-build-tracker-zi5yv9ck\\a26f0086ae0a3a42b2df1119b19e6982d0cc6ba2b3c0a5678897774c'
~~~

macOS with:
~~~
    "_PyUnicode_AsEncodedString", referenced from:
        pybind11::detail::error_fetch_and_normalize::format_value_and_trace() const in _defaults.pic.o
    "_PyUnicode_AsUTF8AndSize", referenced from:
        pybind11::detail::string_caster<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, false>::load(pybind11::handle, bool) in _defaults.pic.o
    "_PyUnicode_AsUTF8String", referenced from:
        pybind11::str::operator std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >() const in _defaults.pic.o
    "_PyUnicode_FromString", referenced from:
        pybind11::detail::make_static_property_type() in _defaults.pic.o
        pybind11::detail::make_default_metaclass() in _defaults.pic.o
        pybind11::detail::make_object_base_type(_typeobject*) in _defaults.pic.o
        pybind11::str::str(char const*) in _defaults.pic.o
        pybind11::detail::dict_getitemstring(_object*, char const*) in _defaults.pic.o
    "_PyWeakref_NewRef", referenced from:
        pybind11::weakref::weakref(pybind11::handle, pybind11::handle) in _defaults.pic.o
    "_Py_GetVersion", referenced from:
        _PyInit__defaults in _defaults.pic.o
        pybind11::detail::type_caster<bool, void>& pybind11::detail::load_type<bool, void>(pybind11::detail::type_caster<bool, void>&, pybind11::handle const&) in _defaults.pic.o
  ld: symbol(s) not found for architecture x86_64
  clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
~~~

macOS probably can be fixed either by passing `-undefined dynamic_lookup` or by linking the Python shared library (see https://github.com/bazelbuild/bazel/issues/16413).

**Comment by traversaro** [comment]
> macOS probably can be fixed either by passing `-undefined dynamic_lookup` or by linking the Python shared library (see [bazelbuild/bazel#16413](https://github.com/bazelbuild/bazel/issues/16413)).

Directly linking the python libraries resulted in a segfault for some reason, let's switch to pass `-undefined dynamic_lookup` .

**Comment by traversaro** [comment]
The Windows problem was fixed by specifying to pip to use a temporary directory inside the working dir.

**Comment by traversaro** [comment]
@conda-forge/help-c-cpp the recipe is ready for review. As I am not expert in bazel I am not sure if I did everything fine from the bazel point of view, so this may be interesting also for @conda-forge/bazel .

File: recipes/labmaze/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{% set name = "labmaze" %}
+{% set version = "1.0.6" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/deepmind/labmaze/archive/6d7e8f058428025cd4253f1ef6a1ed6618d9b0ea.zip
+  sha256: dbd3e0b96e1b50936846bdc6f8f63d1199ca6319aefa64c0f5c9a5d29858f605
```
Comment (carterbox) [inline recipes/labmaze/meta.yaml:10]: https://github.com/deepmind/labmaze/issues/5

File: recipes/labmaze/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - bazel
+    - bazel-toolchain  # [not win]
+
+  host:
+    - pybind11
+    - pybind11-abi
+    - abseil-cpp
```
Comment (carterbox) [inline recipes/labmaze/meta.yaml:29]: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=630409&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=5645

There's linking warnings about libabseil. Are you accidentally static linking?

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/labmaze/meta.yaml (hunk 2 above)
Comment (h-vetinari) [reply recipes/labmaze/meta.yaml:29]: This should be:
```suggestion
    - libabseil
```

File: recipes/labmaze/unvendor_deps.patch [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+ )
+ 
+-http_archive(
++new_local_repository(
+     name = "com_google_absl",
+-    sha256 = "54707f411cb62a26a776dad5fd60829098c181700edcd022ea5c2ca49e9b7ef1",
+-    strip_prefix = "abseil-cpp-20220623.1",
+-    url = "https://github.com/abseil/abseil-cpp/archive/20220623.1.zip",
++    path = "/home/traversaro/mambaforge/envs/labmazedev/include",  # absl placeholder
++    build_file = "@//bazel:system_absl.BUILD"
```
Comment (h-vetinari) [inline recipes/labmaze/unvendor_deps.patch:31]: How can this local path work in CI, or is this simply never built...?

Comment (traversaro) [reply recipes/labmaze/unvendor_deps.patch]: This is replaced by setup.py, following the structure already used in the recipe for the python headers, however given that a random local path was confusing, I substituted it with a clearly fake path.

File: recipes/labmaze/meta.yaml (hunk 2 above)
Comment (traversaro) [reply recipes/labmaze/meta.yaml]: > https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=630409&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=5645
> 
> There's linking warnings about libabseil. Are you accidentally static linking?

I am using just an header-only part of abseil. I added an ignore of libabseil's run_exports in https://github.com/conda-forge/staged-recipes/pull/21512/commits/d85e453f27bd12a5b3bd9460dde0d3bd93845525  to account for that.

File: recipes/labmaze/meta.yaml (hunk 1 above)
Comment (traversaro) [reply recipes/labmaze/meta.yaml]: I updated the recipe to be more clear in https://github.com/conda-forge/staged-recipes/pull/21512/commits/0c0c674e296fc03fcd3200b4e3ad62554fef73c1 .

**Comment by traversaro** [comment]
@h-vetinari @carterbox thanks a lot for the comments, I tried to address them, once the review is completed I will stash the commits because ~35 are quite a lot.

**Pull request merged**